src/Makefile

```diff
@@ -2852,302 +2852,302 @@ auto/gui_gtk_gresources.h: gui_gtk_res.xml $(GUI_GTK_RES_INPUTS)
 # specified for each file separately.
 
 objects:
-	mkdir objects
+	mkdir -p objects
 
-objects/arabic.o: arabic.c
+objects/arabic.o: objects arabic.c
```

Comment:
> Now, each object file target ensures the objects directory is created before creating the `object` file in it.

I think this might be slow. A different approach to consider is to depend on something like `objects/.dirstamp` and create `objects` directory in rules for `objects/.dirstamp` target. This is how automake and probably some other build systems do it.